File: src/main/kotlin/de/gieskerb/class_craft/listener/WorldSaveListener.java
package de.gieskerb.class_craft.listener;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.world.WorldSaveEvent;
import org.jetbrains.annotations.NotNull;

public class WorldSaveListener implements Listener {

    private static String worldSaveName = null;

    @EventHandler
    public void onWorldSaveEvent(@NotNull WorldSaveEvent event) {
        if (worldSaveName == null) {
            worldSaveName = event.getWorld().getName();
        }
        if (event.getWorld().getName().equals(worldSaveName)) {
            // This will only be called once per save not 3 times(for each world once)
            PlayerQuitListener.saveAllPlayersData();
        }
    }

}

File: src/main/kotlin/de/gieskerb/class_craft/listener/WorldSaveListener.kt
package de.gieskerb.class_craft.listener

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.world.WorldSaveEvent

class WorldSaveListener : Listener {
    @EventHandler
    fun onWorldSaveEvent(event: WorldSaveEvent) {
        if (worldSaveName == null) {
            worldSaveName = event.world.name
        }
        if (event.world.name == worldSaveName) {
            // This will only be called once per save not 3 times(for each world once)
            PlayerQuitListener.saveAllPlayersData()
        }
    }

    companion object {
        private var worldSaveName: String? = null
    }
}